interpreter/decoder.h

```diff
@@ -0,0 +1,20 @@
+#ifndef INTERPRETER_DECODER_H
+#define INTERPRETER_DECODER_H
+
+#include "interpreter/instruction.h"
+
+namespace sim {
```

Comment:
Please, rename namespace to 'simulator' or change namespace name to 'sim' in memory folder